cloud/resource_cloud_connection.go

```diff
@@ -0,0 +1,258 @@
+		},
+		Schema: map[string]*schema.Schema{
+			"organization": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  descriptions["organization"],
+				ValidateFunc: validateNotBlank,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  descriptions["name"],
+				ValidateFunc: validateNotBlank,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  descriptions["type"],
+				ValidateFunc: validateNotBlank,
+			},
+			"aws": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: descriptions["aws"],
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"account_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"gcp": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: descriptions["gcp"],
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"project": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceCloudConnectionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	namespace := d.Get("organization").(string)
+	name := d.Get("name").(string)
+	connectionType := d.Get("type").(cloudv1alpha1.ConnectionType)
+	aws := d.Get("aws").([]interface{})
+	gcp := d.Get("gcp").([]interface{})
+	clientSet, err := getClientSet(getFactoryFromMeta(meta))
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("ERROR_INIT_CLIENT_ON_CLOUD_CONNECTION: %w", err))
+	}
+
+	cloudConnection := &cloudv1alpha1.CloudConnection{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "CloudConnection",
+			APIVersion: cloudv1alpha1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+
+		Spec: cloudv1alpha1.CloudConnectionSpec{
+			ConnectionType: connectionType,
+			AWS:            &cloudv1alpha1.AWSCloudConnection{},
+			GCloud:         &cloudv1alpha1.GCloudConnection{},
+		},
+	}
+
+	if len(aws) > 0 {
+		for _, awsItem := range aws {
+			awsMapItem := awsItem.(map[string]interface{})
+			if awsMapItem["account_id"] != nil {
+				accountId := awsMapItem["account_id"].(string)
+				cloudConnection.Spec.AWS.AccountId = accountId
+			}
+		}
+	}
+
+	if len(gcp) > 0 {
+		for _, gCloudItem := range gcp {
+			gCloudItemMap := gCloudItem.(map[string]interface{})
+			if gCloudItemMap["project"] != nil {
+				project := gCloudItemMap["project"].(string)
+				cloudConnection.Spec.GCloud.ProjectId = project
+			}
+		}
+	}
+
+	if cloudConnection.Spec.AWS == nil && cloudConnection.Spec.GCloud == nil {
+		return diag.FromErr(fmt.Errorf("ERROR_CREATE_CLOUD_CONNECTION: " + "One of aws.accountId or gcp.project must be set"))
+	}
+
+	pi, err := clientSet.CloudV1alpha1().CloudConnections(namespace).Create(ctx, cloudConnection, metav1.CreateOptions{
```

Comment:
It seems that pi stands for pulsarinsatnce?

Reply:
Fixed